sympy/polys/matrices/domainmatrix.py

```diff
@@ -30,6 +30,23 @@ def __init__(self, rows, shape, domain):
         self.shape = shape
         self.domain = domain
 
+    def __getitem__(self, key):
+        i, j = key
+        m, n = self.shape
+        if not (isinstance(i, slice) or isinstance(j, slice)):
+            return DomainScalar(self.rep.getitem(i, j), self.domain)
+
+        if not isinstance(i, slice):
+            if i < 0:
+                i = m + i
+            i = slice(i, i+1)
```

Comment:
Here this should raise if `i` is out of range:
```python
In [3]: M = Matrix([[1, 2], [3, 4]])                                                                                              

In [4]: M[:,3]                                                                                                                    
---------------------------------------------------------------------------
IndexError 
```
I would write this as:
```
if not -m <= i < m:
    raise IndexError(...)
i = i % m
i = islice(i, i+1)
```

Reply:
Sure, Done.
Thanks.